pkg/executables/kubectl.go

```diff
@@ -735,3 +735,35 @@ func (k *Kubectl) ValidateNodesVersion(ctx context.Context, kubeconfig string, k
 	}
 	return nil
 }
+
+func (k *Kubectl) ApplyTaint(ctx context.Context, kubeconfig string, node string, taint corev1.Taint) error {
+	params := []string{"taint", "nodes", node, fmt.Sprintf("%s=%s:%s", taint.Key, taint.Value, taint.Effect), "--kubeconfig", kubeconfig}
+	_, err := k.executable.Execute(ctx, params...)
+	return err
+}
+
+func (k *Kubectl) GetWorkerNodes(ctx context.Context, kubeconfig string) ([]string, error) {
+	var workerNodes []string
+	template := "{{range .items}}{{.metadata.name}}\n{{end}}"
+	params := []string{"get", "nodes", "-o", "go-template", "--template", template, "--kubeconfig", kubeconfig}
+	buffer, err := k.executable.Execute(ctx, params...)
+	if err != nil {
+		return workerNodes, err
+	}
+	scanner := bufio.NewScanner(strings.NewReader(buffer.String()))
+	for scanner.Scan() {
+		node := scanner.Text()
+		if len(node) != 0 {
+			params = []string{"get", "node", node, "--kubeconfig", kubeconfig}
+			buffer, err = k.executable.Execute(ctx, params...)
+			if err != nil {
+				return workerNodes, err
+			}
+			output := buffer.String()
+			if !strings.Contains(output, "control-plane") {
```

Comment:
this kind of logic to filter stuff I think first better in `clients/kubectl.go`. We should start moving logic that is exclusive to us in there and leave this for low level operations
But no need to change as part of this PR